nanofi/include/api/nanofi.h

```diff
@@ -71,62 +94,173 @@ typedef int c2_start_callback(char *);
+ * @param processor the processor the property is set for
+ * @param name name of the property
+ * @param value value of the property
+ * @return 0 in case of success, -1 otherwise (the processor doesn't support such property)
+ **/
+int set_property(processor * processor, const char * name, const char * value);
 
-int set_standalone_property(standalone_processor*, const char*, const char *);
+/**
+ * Set property for a standalone processor
+ * @param processor the processor the property is set for
+ * @param name name of the property
+ * @param value value of the property
+ * @return 0 in case of success, -1 otherwise (the processor doesn't support such property)
+ **/
+int set_standalone_property(standalone_processor * processor, const char * name, const char * value);
 
-int set_instance_property(nifi_instance *instance, const char*, const char *);
+/**
+ * Set property for an instance
+ * @param instance the instance the property is set for
+ * @param name name of the property
+ * @param value value of the property
+ * @return 0 in case of success, -1 otherwise. Always succeeds unless instance or name is nullptr/emtpy.
+ **/
+int set_instance_property(nifi_instance *instance, const char * name, const char * value);
 
-char * get_property(const processor_context *  context, const char * name);
+/**
+ * Get a property. Should be used in custom processor logic callbacks.
+ * @attention The returned value transfers ownership, it's the callers responsibility to free it!
+ * @param context the current processor context
+ * @param name name of the property
+ * @return null-terminated char* in case of success, nullptr otherwise
+ **/
+char * get_property(const processor_context * context, const char * name);
 
-int free_flow(flow *);
+/**
+ * Free a flow
+ * @param flow the flow to free
+ * @attention All the processor in the flow are freed, too! Actions performed on freed processors are undefined!
+ * @return 0 in case of success, -1 otherwise. Always succeeds unless flow is nullptr.
+ **/
+int free_flow(flow * flow);
 
+/**
+ * Get the next flow file of the given flow
+ * @param instance the instance the flow belongs to
+ * @param flow the flow to get flowfile from
+ * @return a flow file record or nullptr in case no flowfile was generated by the flow
+ **/
 flow_file_record *get_next_flow_file(nifi_instance *, flow *);
 
-size_t get_flow_files(nifi_instance *, flow *, flow_file_record **, size_t);
+/**
+ * Get all flow files of the given flow
+ * @param instance the instance the flow belongs to
+ * @param flow the flow to get flowfiles from
+ * @param flowfiles target area to copy the flowfiles to
+ * @param size the maximum number of flowfiles to copy to target (size of target)
+ * @return the number of flow files copies to target. Less or equal to size.
+ **/
+size_t get_flow_files(nifi_instance * instance, flow * flow, flow_file_record ** flowfiles, size_t size);
 
 flow_file_record *get(nifi_instance *,flow *, processor_session *);
```

Comment:
This is Python only as well.

Reply:
I think this will be used across languages and very much by custom C processors.

Reply:
I don't think custom C processors can benefit from this at the moment as ptrs to "instance" and "flow" are not available in processor logic callbacks. That's the reason I created "get_flowfile".
Would you mind deprecating this in case Python works with the new one, too?

Reply:
The nomenclature that's common in NiFi is get, so we should work toward keeping some commonality, but get(...) can be changed to use the structs you've defined as it's new within this release and there are no guarantees. I'll create a follow on ticket (MINIFICPP-692). At the very least the consistency should exist be the language python, ruby, or C in how funcitons get accessed for obtaining the flowfile. Naming is secondary to that need.